jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1476,12 +1423,13 @@ bool java_object_factoryt::gen_nondet_enum_init(
   const member_exprt enum_array_expr =
     member_exprt(deref_expr, "data", comps[2].type());
 
-  const symbol_exprt &index_expr = gen_nondet_int_init(
-    assignments,
-    "enum_index_init",
+  const symbol_exprt &index_expr = generate_nondet_int(
     from_integer(0, java_int_type()),
     minus_exprt(length_expr, from_integer(1, java_int_type())),
-    location);
+    "enum_index_init",
+    loc,
```

Comment:
Just like @romainbrenguier I'm surprised this even compiles? As it does, is the `location` parameter for this method unnecessary?

Reply:
See my comment marked :tangerine: above.
The way the object factory passes variables around is currently quite messy in general, it uses a mix of
- class fields (even though there isn't a publicly visible class here; from what I can see the class is only used to store fields)
- many parameters to most functions
- a poorly named `java_object_factory_parameterst` which groups further arguments into a struct.
The three common ways of moving lots of variables around are all combined in one place here... I would like to generally tidy this up a bit, probably moving away from lots of function parameters and fields to bundling more things in (a) struct(s). Let me know what you think.